docs/pipelines.md

```diff
@@ -0,0 +1,149 @@
+# Pipelines
+
+This document defines `Pipelines` and their capabilities.
+
+---
+
+- [Syntax](#syntax)
+  - [Declared resources](#declared-resources)
+  - [Pipeline Tasks](#pipeline-tasks)
+    - [From](#from)
+- [Examples](#examples)
+
+## Syntax
+
+To define a configuration file for a `Pipeline` resource, you can specify the
+following fields:
+
+- Required:
+  - [`apiVersion`][kubernetes-overview] - Specifies the API version, for example
+    `pipeline.knative.dev/v1alpha1`.
+  - [`kind`][kubernetes-overview] - Specify the `Pipeline` resource object.
+  - [`metadata`][kubernetes-overview] - Specifies data to uniquely identify the
+    `Pipeline` resource object, for example a `name`.
+  - [`spec`][kubernetes-overview] - Specifies the configuration information for
+    your `Pipeline` resource object. In order for a `Pipeline` to do anything, the
+    spec must include:
+    - [`tasks`](#pipeline-tasks) - Specifies which `Tasks` to run and how to run them
+- Optional:
+  - [`resources`](#declared-resources) - Specifies which [`PipelineResources`](resources.md)
+    of which types the `Pipeline` will be using in its [Tasks](#pipeline-tasks)
+  - [`timeout`](#timeout) - Specifies timeout after which the `Pipeline` will fail.
+
+[kubernetes-overview]:
+  https://kubernetes.io/docs/concepts/overview/working-with-objects/kubernetes-objects/#required-fields
+
+### Declared resources
+
+In order for a `Pipeline` to interact with the outside world, it will probably need
+[`PipelineResources`](#creating-pipelineresources) which will be given to
+`Tasks` as inputs and outputs.
+
+Your `Pipeline` must declare the `PipelineResources` it needs in a `resources`
+section in the `spec`, giving each a name which will be used to refer to these
+`PipelineResources` in the `Tasks`.
+
+For example:
+
+```yaml
+spec:
+  resources:
+    - name: my-repo
+      type: git
+    - name: my-image
+      type: image
+```
+
+### Pipeline Tasks
+
+A `Pipeline` will execute a sequence of [`Tasks`](tasks.md) in the order they are declared in.
+At a minimum, this declaration must include a reference to the `Task`:
+
+```yaml
+  tasks:
+    - name: build-the-image
+      taskRef:
+        name: build-push
+```
+
+[Declared `PipelineResources`](#declared-resources) can be given to `Task`s in the `Pipeline` as
+inputs and outputs, for example:
+
+```yaml
+spec:
+  tasks:
+    - name: build-the-image
+      taskRef:
+        name: build-push
+      resources:
+        inputs:
+          - name: workspace
+            resource: my-repo
+        outputs:
+          - name: image
+            resource: my-image
+```
+
+The resource `my-image` is expected to be given to the `deploy-app` `Task` from
+the `build-app` `Task`. This means that the `PipelineResource` `my-image` must
+also be declared as an output of `build-app`.
+
```

Comment:
is that the same as `build-the-image` in the example above?

Reply:
whoa this block seems completely out of place 😊